### PR TITLE
feat(*): role property to icon, focusable remove chip

### LINKED ISSUE #480: Chip Remove icon
## Description  
The remove icon of the chip can't be reached with a keyboard. 

## Expected result  
The remove icon of the chip should be reachable with a keyboard.

### PR DESCRIPTION
**Pull request by ChronosSF**
Closes #480

### CODE REVIEW

File: src/components/icon/icon.ts [hunk 1]
```diff
@@ -71,14 +71,22 @@ export default class IgcIconComponent extends SizableMixin(LitElement) {
...
+  public role = 'img';
+
   constructor() {
     super();
     this.size = 'medium';
   }
 
   public override connectedCallback() {
     super.connectedCallback();
-    this.setAttribute('role', 'img');
```
Comment (rkaraivanov) [inline src/components/icon/icon.ts:81]: Can we check for an existing role attribute and then set the default if none is found?

Something about a public property used in an internal context that is reflected as a standard attribute feels off to me.

**Review by rkaraivanov — CHANGES_REQUESTED** [review]

**Comment by rkaraivanov** [comment]
@ChronosSF
 
While this PR fixes the focus issue, triggering an "activation" event through <kbd>Space</kbd> | <kbd>Enter</kbd> on the remove button does not fire the **igcRemove** event.
It does it on click, so this also falls under general accessibility. 

I suggest we fix that too either in this PR or in a new one.

File: src/components/chip/chip.ts [hunk 2]
```diff
@@ -112,12 +118,16 @@ export default class IgcChipComponent extends SizableMixin(
             ? html`<slot
                 @slotchange=${this.slotChanges}
                 @click=${this.handleRemove}
+                @keyup=${this.handleRemove}
```
Comment (rkaraivanov) [inline src/components/chip/chip.ts:121]:
```suggestion
                @keyup=${this.handleKeyup}
```

File: src/components/icon/icon.ts [hunk 3]
```diff
@@ -78,7 +78,9 @@ export default class IgcIconComponent extends SizableMixin(LitElement) {
...
+    if (!this.getAttribute('role')) {
```
Comment (rkaraivanov) [inline src/components/icon/icon.ts:81]:
```suggestion
    if (!this.hasAttribute('role')) {
```

File: stories/icon.stories.ts [hunk 4]
```diff
@@ -33,6 +33,12 @@ const metadata = {
       control: 'boolean',
       defaultValue: false,
     },
+    role: {
```
Comment (rkaraivanov) [inline stories/icon.stories.ts:36]: Can you regenerate the icon stories and commit them as well?

File: src/components/chip/chip.ts [hunk 5]
```diff
@@ -78,6 +78,12 @@ export default class IgcChipComponent extends SizableMixin(
...
   protected handleRemove(e: Event) {
+    if (
+      e.type === 'keyup' &&
+      ![' ', 'enter'].includes((e as KeyboardEvent).key.toLowerCase())
+    ) {
+      return;
+    }
     this.emitEvent('igcRemove');
     e.stopPropagation();
   }
```
Comment (rkaraivanov) [inline src/components/chip/chip.ts:89]:
```suggestion
  protected handleRemove(e: Event) {
    this.emitEvent('igcRemove');
    e.stopPropagation();
  }
  
  protected handleKeyup(e: KeyboardEvent) {
    if (/\s|enter/i.test(e.key)) {
      this.handleRemove(e);
    }
  }
```
A bit cleaner IMO

**Review by rkaraivanov — CHANGES_REQUESTED** [review]

File: src/components/chip/chip.ts (hunk 5 above)
Comment (ChronosSF) [reply src/components/chip/chip.ts:89]: Thought regex is evil ^^

Comment (rkaraivanov) [reply src/components/chip/chip.ts:89]: It is a tool. It can be used for both good and evil 😄

**Review by rkaraivanov — APPROVED** [review]

**Pull request merged**